bindModel.js

```diff
@@ -45,8 +45,18 @@ var inputTypeBindings = {
   select: function (attributes, children, binding) {
     var currentValue = binding.get()
 
-    var options = children.filter(function (child) {
-      return child.tagName && child.tagName.toLowerCase() === 'option'
+    var options = []
+    children.forEach(function (child) {
+      if (child.tagName && child.tagName.toLowerCase() === 'optgroup') {
+        return child.children.filter(function (optChild) {
```

Comment:
bit confused by this `return` here and the use of `filter`. The surrounding `forEach` isn't going to use the result of the `filter` which makes me think that it could be written with `forEach` instead of `filter`. Also I think it reads better if we do:

```js
if (optgroup) {
   ...
} else if (option) {
   ...
}
```

Rather than returning from the first `if` - even if it does the same thing. if else if makes me think there's only two options, whereas a return in the first if makes me think it's more complicated than that... Maybe just me.

Reply:
I think the reason for that is mostly because it was late at night and I wasn't thinking straight.